### ISSUE TITLE
[BUG] 1.19.2 paper plugin won't stop mob spawning

### ISSUE THREAD

**Issue by cosmin1805** [TheDutchMC04/HaroTorch#18]
**Describe the bug**
I crafted a torch and placed it but mobs keep spawning. No errors in the console, I don't know if maybe one of my plugins is blocking it. I also wanted to add the blaze to be excluded and this was not working:
mobsExcludeFromBlockList:
- "blaze"
And the console was displaying this:
[HaroTorch-V2] Provided mob type blaze is not valid. Please check your configuration file!

Plugin list:
![image](https://user-images.githubusercontent.com/93390391/209438023-a3a4131f-daac-457a-b3c5-817ac11ab99c.png)

**Version**
Minecraft version: 1.19.2
Spigot/Paper/Bukkit version [/version]: This server is running Paper version git-Paper-307 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: 476ef25)
You are running the latest version
HaroTorch version: 2.4.4

**Comment by TobiasDeBruijn** [comment]
Hm, another paper one.

Could you give regular old spigot a try? We've had some weirdness with Paper in the past, curious if that's back.

**Comment by cosmin1805** [comment]
Well, nope mobs still spawn even on spigot, it solved just the blaze problem.

**Comment by cosmin1805** [comment]
Could it be because of worldguard?

**Comment by TobiasDeBruijn** [comment]
That's certainly a possibility. Try removing it, see if that fixes it.

**Comment by cosmin1805** [comment]
Unfortunately still not working, any idea what could be making it not work?

**Comment by cosmin1805** [comment]
Well, I removed all the plugins and tried both paper and spigot. And mobs still spawn in the radius of the torch.

**Comment by TobiasDeBruijn** [comment]
Hm, interesting.

That is all with the min/max set to -1? Could you try setting it to 255?

**Comment by cosmin1805** [comment]
Also I just realized something would this plugin work even if the server is in offline mode?

**Comment by TobiasDeBruijn** [comment]
It should, yes.

**Comment by cosmin1805** [comment]
> 255

Still not working :(

**Comment by cosmin1805** [comment]
Ok, so after looking throw your code when you check if an entity is in range of a torch. When the first torch out of all of them is not in range you return false and don't check the other torches. This is how i fixed this, but i don't know how to use github so i pasting it here.
```
private boolean torchInRange(Location entityLocation) {
		int yAboveLim = this.plugin.getConfigManifest().torchAboveYRange;
		int yBelowLim = this.plugin.getConfigManifest().torchBelowYRange;
		boolean inRange = false;
		//Iterate over all Torches
		for(Torch t : TorchHandler.getTorches()) {
			
			//Check if the spawn occured in the same World as the current Torch
			if(!t.getLocation().getWorld().equals(entityLocation.getWorld())) continue;
			
			if(this.plugin.getConfigManifest().getTorchRangeShape() == TorchRangeShape.CIRCLE) {
				//Check if the distance cylindrical is less than the defined range squared
				if(TorchHandler.getDistanceCylindrical(t.getLocation(), entityLocation) > HaroTorch.RANGE) {
					continue;
				}
			} else {
				Location lTorch = t.getLocation();
				double distanceX = Math.abs(lTorch.getX() - entityLocation.getX());
				double distanceZ = Math.abs(lTorch.getZ() - entityLocation.getZ());
				
				int radius = this.plugin.getConfigManifest().torchRange;
				if(!(distanceX < radius && distanceZ < radius)) {
					continue;
				}	
			}
			
			boolean yAboveSatisfied = false;
			boolean yBelowSatisfied = false;
			
			if(yAboveLim != -1) {
				if(t.getLocation().getY() + yAboveLim <= entityLocation.getY()) {
					yAboveSatisfied = true;
				}
			} else {
				yAboveSatisfied = true;
			}
			
			if(yBelowLim != -1) {
				if(t.getLocation().getY() - yBelowLim >= entityLocation.getY()) {
					yBelowSatisfied = true;
				}
			} else {
				yBelowSatisfied = true;
			}
			
			inRange = yAboveSatisfied && yBelowSatisfied;
		}
		
		return inRange;
	}
```

**Comment by TobiasDeBruijn** [comment]
Fixed in v2.5.0
